File: 0922-sort-array-by-parity-ii/0922-sort-array-by-parity-ii.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        int n = nums.size();
        int evenIndex = 0;
        int oddIndex = 1;

        while (evenIndex < n && oddIndex < n) {
            while (evenIndex < n && nums[evenIndex] % 2 == 0) {
                evenIndex += 2;
            }

            while (oddIndex < n && nums[oddIndex] % 2 != 0) {
                oddIndex += 2;
            }

            if (evenIndex < n && oddIndex < n) {
                swap(nums[evenIndex], nums[oddIndex]);
            }
        }
        return nums;
    }
};

File: 0922-sort-array-by-parity-ii/0922-sort-array-by-parity-ii.java
class Solution {
    public int[] sortArrayByParityII(int[] nums) {
        
        int n = nums.length;
        int[] arr = new int[n];
        int[] newArr = new int[n];
        int start = 0;
        int last = arr.length-1;
        
        for(int i=0;i<n;i++){
            if(nums[i]%2 == 0){
                arr[start]= nums[i];
                start++;
            }else{
                arr[last] = nums[i];
                last--;
            } 
        }
        
        int s = 0;
        int l = arr.length-1;
        for(int i=0;i<n;i++){
            if(i%2 == 0){
                newArr[i]=arr[s];
                s++;
            }else{
                newArr[i]=arr[l];
                l--;
            }
            
        }
        
        return newArr;
    }
}